addons/point_of_sale/static/src/js/models.js

```diff
@@ -2145,11 +2140,24 @@ exports.Order = Backbone.Model.extend({
             this.pos.db.save_unpaid_order(this);
         }
     },
+    /**
+     * Initialize PoS order from a JSON string.
+     *
+     * If the order was created in another session, the sequence number should be changed so it doesn't conflict
+     * with orders in the current session.
+     * Else, the sequence number of the session should follow on the sequence number of the loaded order.
+     *
+     * @param {object} json JSON representing one PoS order.
+     */
     init_from_JSON: function(json) {
         var client;
-        this.sequence_number = json.sequence_number;
-        this.pos.pos_session.sequence_number = Math.max(this.sequence_number+1,this.pos.pos_session.sequence_number);
-        this.session_id = json.pos_session_id;
+        if (json.pos_session_id !== this.pos.pos_session.id) {
+            this.sequence_number = this.pos.pos_session.sequence_number++;
```

Comment:
With the logic here, we cannot guarantee that two orders don't have the same sequence number (you could load one from previous session which lead to give it the sequence_number '1', and then load the json of the order '1') which lead to two orders with the same sequence_number.

We must first push the orders of the current session, then the ones from the older session.

Reply:
@pimodoo You are right, I changed it in the load_orders function :+1: